src/Media_Command.php

```diff
@@ -279,15 +286,34 @@ function import( $args, $assoc_args = array() ) {
 				$post_array['post_title'] = preg_replace( '/\.[^.]+$/', '', Utils\basename( $file ) );
 			}
 
-			// Deletes the temporary file.
-			$success = media_handle_sideload( $file_array, $assoc_args['post_id'], $assoc_args['title'], $post_array );
-			if ( is_wp_error( $success ) ) {
-				WP_CLI::warning( sprintf(
-					"Unable to import file '%s'. Reason: %s",
-					$orig_filename, implode( ', ', $success->get_error_messages() )
-				) );
-				$errors++;
-				continue;
+			if ( \WP_CLI\Utils\get_flag_value( $assoc_args, 'skip-copy' ) ) {
+				$wp_filetype = wp_check_filetype( $file, null );
+				$post_array['post_mime_type'] = $wp_filetype['type'];
+				$post_array['post_status'] = 'inherit';
+
+				$success = wp_insert_attachment( $post_array, $file, $assoc_args['post_id'] );
+				if ( ! is_wp_error( $success ) ) {
```

Comment:
You should switch the conditions and bail early on error.

So, instead of this:
```
if ( ! is_wp_error( $success ) ) {
    // Handle success.
}
if ( is_wp_error( $success ) ) {
    // Handle failure.
    continue;
}
```
you should prefer something like this:
```
if ( is_wp_error( $success ) ) {
    // Handle failure.
    continue;
}
// Handle success.
```